### PR TITLE
Run ILVerify

### PR DESCRIPTION
**Pull request by vzarytovskii**
Run ilverify on each PR. Currently it fails on a bunch of "invalid" codegen. Not sure if it makes sense to ignore current one, or create a baseline (which would be much better).

I will see how it runs in CI and create a baseline, so no new codegen changes are introduced.

### CODE REVIEW

File: eng/ilverify.bsl [hunk 1]
```diff
@@ -0,0 +1,555 @@
+[IL]: Error [StackUnexpected]: : Internal.Utilities.Collections.HashMultiMap`2::.ctor(int32, [S.P.CoreLib]System.Collections.Generic.IEqualityComparer`1<!0>, [FSharp.Core]Microsoft.FSharp.Core.FSharpOption`1<bool>)][offset 0x00000060][found ref 'object'][expected ref '[S.P.CoreLib]System.Collections.Generic.IDictionary`2<T0,Microsoft.FSharp.Collections.FSharpList`1<T1>>'] Unexpected type on the stack.
```
Comment (vzarytovskii) [inline eng/ilverify.bsl:1]: These are all issues we have currently in `FSharp.Core` and `FSharp.Compiler.Service` when building with `Proto` compiler.

File: eng/ilverify.ps1 [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        $parts = $_ -split " "
+        "$($parts[0]) $($parts[1]) $($parts[2]) $($parts[4..$parts.Length])"
+    } elseif ($_ -match "Error\(s\) Verifying") {
+        # do nothing
+    } else {
+        $_
+    }
+}
+
+$baseline_file = Join-Path $repo_path "eng" "ilverify.bsl"
```
Comment (T-Gro) [inline eng/ilverify.ps1:116]: The .bsl is common, but il-verify runs per project and configuration?
Can we then tell from which source an entry comes from? e.g. difference between ns20 and net90...

ilverify_output could already be maintained per run, to allow for .bsl granularity.

Comment (T-Gro) [reply eng/ilverify.ps1:116]: ilverify_$project_$configuration_$tfm.bsl or similar?

>

File: eng/ilverify.bsl [hunk 3]
```diff
@@ -0,0 +1,555 @@
...
+[IL]: Error [UnmanagedPointer]: : FSharp.Compiler.IO.SafeUnmanagedMemoryStream::.ctor(uint8*, int64, int64, [S.P.CoreLib]System.IO.FileAccess, object)][offset 0x00000001] Unmanaged pointers are not a verifiable type.
+[IL]: Error [UnmanagedPointer]: : FSharp.Compiler.IO.RawByteMemory::.ctor(uint8*, int32, object)][offset 0x00000009] Unmanaged pointers are not a verifiable type.
+[IL]: Error [StackByRef]: : FSharp.Compiler.IO.RawByteMemory::get_Item(int32)][offset 0x0000001E][found Native Int] Expected ByRef on the stack.
+[IL]: Error [StackByRef]: : FSharp.Compiler.IO.RawByteMemory::set_Item(int32, uint8)][offset 0x00000025][found Native Int] Expected ByRef on the stack.
+[IL]: Error [ReturnPtrToStack]: : Internal.Utilities.Text.Lexing.LexBuffer`1::get_LexemeView()][offset 0x00000019] Return type is ByRef, TypedReference, ArgHandle, or ArgIterator.
+[IL]: Error [StackUnexpected]: : Internal.Utilities.Text.Lexing.UnicodeTables::scanUntilSentinel([FSharp.Compiler.Service]Internal.Utilities.Text.Lexing.LexBuffer`1<char>, int32)][offset 0x00000079][found Short] Unexpected type on the stack.
```
[the remark this answers is not in the document]
Comment (vzarytovskii) [reply eng/ilverify.bsl:9]: 100%, this is just issues which been there for quite some time, and are there to not fail the build

File: eng/ilverify.ps1 (hunk 2 above)
Comment (vzarytovskii) [reply eng/ilverify.ps1:116]: Funny enough, paths were there, but I had to cut them, because they will be different on different machines

Comment (vzarytovskii) [reply eng/ilverify.ps1]: Separated baselines btw

**Comment by vzarytovskii** [comment]
Separated baselines, if it succeeds, it's ready.

**Review by T-Gro — APPROVED** [review]

**Comment by T-Gro** [comment]
The CI step feels wrong - it finished in 1m30s only, I think it did not do the build at all.
And has some warnings but still has to failed CI.

**Review by T-Gro — CHANGES_REQUESTED** [review]
See CI - was too fast and did not error.

**Comment by vzarytovskii** [comment]
> See CI - was too fast and did not error.

Ah, path issue, gonna fix it tomorrow

**Comment by vzarytovskii** [comment]
Wasn't the path issue, but the issue of me commenting out part of the script to test locally, and forgetting to uncomment. Script will also fail now if the DLL is not on the path.

**Comment by vzarytovskii** [comment]
@T-Gro this is passing now:
image


Also checked with false positives (invalid baselines) - resulted in the step failure.

**Review by 0101 — APPROVED** [review]

**Pull request merged**